VenafiPS/Public/Get-TppEngineFolder.ps1

```diff
@@ -0,0 +1,106 @@
+<#
+.SYNOPSIS
+Get TPP folder/engine assignments
+.DESCRIPTION
+When the input is a policy folder, retrieves an array of assigned TPP processing engines.
+When the input is a TPP engine, retrieves an array of assigned policy folders.
+If there are no matching assignments, nothing will be returned.
+.PARAMETER Path
+The full DN path to a TPP processing engine or policy folder.
+.PARAMETER InputObject
+TPPObject belonging to the 'Venafi Platform' or 'Policy' class.
+.PARAMETER Guid
+Guid of a TPP processing engine or policy folder.
+.PARAMETER VenafiSession
+Authentication for the function.
+The value defaults to the script session object $VenafiSession created by New-VenafiSession.
+A TPP token can also provided, but this requires an environment variable TPP_SERVER to be set.
+.INPUTS
+Path, Guid, TppObject
+.OUTPUTS
+TppObject[]
+.EXAMPLE
+Get-TppEngineFolder -Path '\VED\Engines\MYVENSERVER'
+Get an array of policy folders assigned to the TPP processing engine 'MYVENSERVER'.
+.EXAMPLE
+Get-TppEngineFolder -Path '\VED\Policy\Certificates\Web Team'
+Get an array of TPP processing engines assigned to the policy folder '\VED\Policy\Certificates\Web Team'.
+.EXAMPLE
+[guid]'866e1d59-d5d2-482a-b9e6-7bb657e0f416' | Get-TppEngineFolder
+When the GUID is assigned to a TPP processing engine, returns an array of assigned policy folders.
+When the GUID is assigned to a policy folder, returns an array of assigned TPP processing engines.
+Otherwise nothing will be returned.
+.LINK
+http://VenafiPS.readthedocs.io/en/latest/functions/Get-TppEngineFolder/
+.LINK
+https://github.com/Venafi/VenafiPS/blob/main/VenafiPS/Public/Get-TppEngineFolder.ps1
+.LINK
+https://docs.venafi.com/Docs/current/TopNav/Content/SDK/WebSDK/r-SDK-GET-ProcessingEngines-Engine-eguid.php
+.LINK
+https://docs.venafi.com/Docs/current/TopNav/Content/SDK/WebSDK/r-SDK-GET-ProcessingEngines-Folder-fguid.php
+#>
+function Get-TppEngineFolder
+{
+    [CmdletBinding()]
+
+    param (
+        [Parameter(Mandatory, ParameterSetName = 'ByPath', ValueFromPipeline, ValueFromPipelineByPropertyName, Position = 0)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript( {
+            if ( $_ | Test-TppDnPath ) { $true }
+            else { throw "'$_' is not a valid DN path" }
+        })]
+        [Alias('DN')]
+        [String] $Path,
+
+        [Parameter(ParameterSetName = 'ByObject', ValueFromPipeline)]
+        [TppObject] $InputObject,
+
+        [Parameter(Mandatory, ParameterSetName = 'ByGuid', ValueFromPipeline)]
+        [ValidateNotNullOrEmpty()]
+        [Alias('ObjectGuid')]
+        [guid] $Guid,
```

Comment:
I like the guid because they tend to be more efficient in some cases and more reference stable in almost all cases. Still, the reality is I almost always end up converting to TppObject regardless so it's probably better to lose it here for consistency.

Just don't take away my guid from Get-TppObject so I can still do a fallback lookup when my name-tweaking admins get overly zealous with my policy folders lol.